khelo/lib/ui/flow/leaderboard/leaderboard_screen.dart

```diff
@@ -0,0 +1,240 @@
+      body: Builder(
+        builder: (context) => _body(context, state),
+      ),
+    );
+  }
+
+  Widget _body(BuildContext context, LeaderboardViewState state) {
+    return Padding(
+      padding: context.mediaQueryPadding,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          _tabView(context),
+          _content(context, state),
+        ],
+      ),
+    );
+  }
+
+  Widget _emptyLeaderboard(BuildContext context) {
+    return EmptyScreen(
+      title: context.l10n.leaderboard_empty_title,
+      description: context.l10n.leaderboard_empty_description,
+      isShowButton: false,
+    );
+  }
+
+  Widget _tabView(BuildContext context) {
+    final tabs = LeaderboardField.values;
+
+    return SingleChildScrollView(
+      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 9),
+      scrollDirection: Axis.horizontal,
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.start,
+        children: List.generate(
+          tabs.length,
+          (index) => Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 4),
+            child: TabButton(
+              tabs[index].getString(context),
+              onTap: () => _controller.jumpToPage(index),
+              selected: index == _selectedTab,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _content(BuildContext context, LeaderboardViewState state) {
+    return Expanded(
+      child: PageView(
+        controller: _controller,
+        onPageChanged: notifier.onTabChange,
+        children: LeaderboardField.values
+            .map((field) => _playerListView(context, state, field))
+            .toList(),
+      ),
+    );
+  }
+
+  Widget _playerListView(
+    BuildContext context,
+    LeaderboardViewState state,
+    LeaderboardField field,
+  ) {
+    final players = field == LeaderboardField.batting
+        ? state.battingLeaderboard
+        : field == LeaderboardField.bowling
+            ? state.bowlingLeaderboard
+            : state.fieldingLeaderboard;
+    if (state.loading) {
+      return const Center(child: AppProgressIndicator());
+    } else if (state.error != null) {
+      return ErrorScreen(
+        error: state.error,
+        onRetryTap: notifier.loadLeaderboard,
+      );
+    } else if (players.isEmpty) {
+      return _emptyLeaderboard(context);
+    }
+
+    return ListView.separated(
+      padding: EdgeInsets.all(16),
+      itemCount: players.length + 1,
+      itemBuilder: (context, index) {
+        if (index < players.length) {
+          final player = players[index];
+          return _leaderboardCell(
+            player: player,
+            field: field,
+            rank: index + 1,
+          );
+        } else {
+          return OnVisibleCallback(
+            onVisible: () => runPostFrame(notifier.loadLeaderboard),
+            child: state.loading
```

Comment:
I think if you will try to load first time then two time show progress indication so please add more condition like `state.loading && players.isNotEmpty`.